File: dart/kitchen_manager/kitchen_manager.dart
import 'chef.dart';
import 'dishwasher.dart';
import 'waiter.dart';

class KitchenManager {
  final Chef _chef = Chef();
  final Dishwasher _dishwasher = Dishwasher();
  final Waiter _waiter = Waiter();

  void run() {
    _chef.prepareFood();
    _waiter.serveCustomers();
    _dishwasher.washDishes();
  }
}

File: typescript/kitchen_manager/KitchenManager.ts
import { Chef } from './Chef';
import { Dishwasher } from './Dishwasher';
import { Waiter } from './Waiter';

export class KitchenManager {
  private chef: Chef = new Chef();
  private dishwasher: Dishwasher = new Dishwasher();
  private waiter: Waiter = new Waiter();

  public run(): void {
    this.chef.prepareFood();
    this.waiter.serveCustomers();
    this.dishwasher.washDishes();
  }
}
